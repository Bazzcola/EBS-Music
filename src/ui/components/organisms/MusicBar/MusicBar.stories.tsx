import React from 'react';
import { MusicBar } from 'ui/components/organisms/MusicBar/MusicBar';
export default {
  title: 'Organisms/MusicBar',
  component: MusicBar
};

export const MusicBar1 = () => {
  return <MusicBar />;
};
