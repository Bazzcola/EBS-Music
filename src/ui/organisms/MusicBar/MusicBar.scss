.music_container{
    display:flex;
    justify-content: space-between;
    padding: 20px 10px 10px 10px;
    border-top:1px #d1d1d6 solid;
    width: 900px;
    .player_buttons_box{
        display:flex;
        align-items: center;
    }
    .control_buttons_box{
        display:flex;
        align-items: center;
    }
}