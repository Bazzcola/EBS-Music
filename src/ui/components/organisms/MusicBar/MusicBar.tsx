import React from 'react';
import { PlayerButtons } from 'ui/components/molecules/PlayerButtons/PlayerButtons';
import { ProgressBar } from 'ui/components/molecules/ProgressBar/ProgressBar';
import { ControlButtons } from 'ui/components/molecules/ControlButtons/ControlButtons';
import 'ui/components/organisms/MusicBar/MusicBar.scss';

export const MusicBar = () => {
  return (
    <div className="music_container">
      <div className="player_buttons_box">
        <PlayerButtons />
      </div>
      <div className="progress_bar_box">
        <ProgressBar />
      </div>
      <div className="control_buttons_box">
        <ControlButtons />
      </div>
    </div>
  );
};
