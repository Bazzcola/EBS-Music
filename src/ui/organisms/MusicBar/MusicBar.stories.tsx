import React from 'react';
import { MusicBar} from 'ui/organisms/MusicBar/MusicBar';
export default {
  title: 'Organisms/MusicBar',
  component: MusicBar
};

export const MusicBar1 = () => {
  return <MusicBar />;
};