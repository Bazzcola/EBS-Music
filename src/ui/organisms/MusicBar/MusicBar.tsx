import React from 'react';
import {PlayerButtons} from 'ui/molecules/PlayerButtons/PlayerButtons';
import {ProgressBar} from 'ui/molecules/ProgressBar/ProgressBar';
import {ControlButtons} from 'ui/molecules/ControlButtons/ControlButtons';
import 'ui/organisms/MusicBar/MusicBar.scss';

export const MusicBar = () => {

    return(
        <div className="music_container">
            <div className="player_buttons_box">
                <PlayerButtons />
            </div>
            <div className="progress_bar_box">
                <ProgressBar />
            </div>
            <div className="control_buttons_box">
                <ControlButtons />
            </div>
        </div>
    )
}