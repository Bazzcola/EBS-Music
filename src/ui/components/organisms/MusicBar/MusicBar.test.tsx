import React from 'react';
import { render } from '@testing-library/react';
import { MusicBar } from './MusicBar';

it('renders MusicBar correctly', () => {
  const { asFragment } = render(<MusicBar />);
  expect(asFragment()).toMatchSnapshot();
});
