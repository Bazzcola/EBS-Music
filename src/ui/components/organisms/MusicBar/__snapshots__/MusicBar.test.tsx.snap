// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MusicBar correctly 1`] = `
<DocumentFragment>
  <div
    class="music_container"
  >
    <div
      class="player_buttons_box"
    >
      <div
        class="btn_box"
      >
        <div
          class="btn_box__item"
        >
          <button
            class="prev_button "
          >
            <svg
              class="svg_prevbtn"
              viewBox="0 0 12 12"
            >
              <path
                d="M2 5.397V.5a.5.5 0 1 0-1 0v11a.5.5 0 1 0 1 0V6.597L11 12V0L2 5.397z"
              />
            </svg>
          </button>
        </div>
        <div
          class="btn_box__item"
        >
          <button
            class="play_button"
          >
            <svg
              class="svg_playbtn"
              viewBox="0 0 12 12"
            >
              <path
                d="M2.5.5v11l9-5.5z"
              />
            </svg>
          </button>
        </div>
        <div
          class="btn_box__item"
        >
          <button
            class="next_button "
          >
            <svg
              class="svg_nextbtn"
              viewBox="0 0 12 12"
            >
              <path
                d="M11 .5v11c0 .276-.232.5-.5.5a.503.503 0 0 1-.5-.5V7.08L1.562 11.95A.375.375 0 0 1 1 11.624V.376c0-.29.312-.47.562-.325L10 4.919V.5c0-.276.232-.5.5-.5.276 0 .5.229.5.5z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="progress_bar_box"
    >
      <div
        class="progress_bar"
      >
        <audio
          autoplay=""
          class="audio_source"
          id="audio"
          preload="metadata"
          src=""
        />
        <div
          class="song_text"
        >
          <span
            class="song_group"
          >
             
          </span>
          <span
            class="song_name"
          />
        </div>
        <div
          class="bar"
        >
          <span
            class="bar__time"
          >
            0:00
          </span>
          <div
            class="bar__progress"
            currentitem="false"
            data-for="bar_time"
            data-tip=""
          >
            <span
              class="bar__progress__knob"
              style="left: -2%;"
            />
            <div
              class="__react_component_tooltip td73cfefc-35ce-43dc-b397-7c0136c03441 place-top type-dark"
              data-id="tooltip"
              id="bar_time"
            >
              <style>
                
  	.td73cfefc-35ce-43dc-b397-7c0136c03441 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.td73cfefc-35ce-43dc-b397-7c0136c03441.place-top {
        margin-top: -10px;
    }
    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-top::before {
        border-top: 8px solid transparent;
    }
    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-bottom {
        margin-top: 10px;
    }
    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-left {
        margin-left: -10px;
    }
    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-left::before {
        border-left: 8px solid transparent;
    }
    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-right {
        margin-left: 10px;
    }
    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-right::before {
        border-right: 8px solid transparent;
    }
    .td73cfefc-35ce-43dc-b397-7c0136c03441.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
              </style>
              0:00
            </div>
          </div>
          <span
            class="bar__time"
          >
            0:00
          </span>
        </div>
      </div>
    </div>
    <div
      class="control_buttons_box"
    >
      <div
        class="control_bar"
      >
        <div
          class="control_buttons"
        >
          <div
            class="control_repeat_button"
          >
            <button
              class="repeat_button"
              currentitem="false"
              data-for="repeat"
              data-tip=""
            >
              <svg
                class="'svg_repeat' svg_repeat"
                viewBox="0 0 12 12"
              >
                <path
                  d="M8.667 9.214H8V10.5L5.333 8.571V8.25c0-.178.149-.321.334-.321h3c1.104 0 2-.864 2-1.929 0-1.066-.896-1.929-2-1.929h-.334A.327.327 0 0 1 8 3.75v-.643c0-.178.148-.321.333-.321h.334C10.507 2.786 12 4.225 12 6s-1.493 3.214-3.333 3.214zM6.333 4.071h-3c-1.105 0-2 .863-2 1.929 0 1.065.895 1.929 2 1.929h.334c.183 0 .333.143.333.321v.643c0 .177-.15.321-.333.321h-.334C1.493 9.214 0 7.775 0 6s1.493-3.214 3.333-3.214H4V1.5l2.667 1.929v.321c0 .177-.15.321-.334.321z"
                />
              </svg>
            </button>
            <div
              class="__react_component_tooltip t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf place-top type-dark"
              data-id="tooltip"
              id="repeat"
            >
              <style>
                
  	.t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-top {
        margin-top: -10px;
    }
    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-top::before {
        border-top: 8px solid transparent;
    }
    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-bottom {
        margin-top: 10px;
    }
    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-left {
        margin-left: -10px;
    }
    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-left::before {
        border-left: 8px solid transparent;
    }
    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-right {
        margin-left: 10px;
    }
    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-right::before {
        border-right: 8px solid transparent;
    }
    .t1396a2d4-8c37-4b43-8d85-e5f3fd1f8caf.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
              </style>
              <span
                class="repeat_tooltip"
              >
                Repeat all track in list
              </span>
            </div>
          </div>
          <div
            class="control_shuffle_button"
          >
            <button
              class="shuffle_button"
              currentitem="false"
              data-for="shuffle"
              data-tip=""
            >
              <svg
                class="svg_shuffle"
                viewBox="0 0 12 12"
              >
                <path
                  d="M9.333 9.214h-.666C6.826 9.214 5.333 7.775 5.333 6c0-1.066-.894-1.929-2-1.929H.683C.298 4.071 0 3.784 0 3.43c0-.356.298-.643.667-.729.137 0 2.65.01 2.666 0 1.841.086 3.334 1.525 3.334 3.3 0 1.065.896 1.929 2 1.929h3c.185 0 .333.143.333.321v.321L9.333 10.5V9.214zm-.666-5.143c-.367 0-.708.103-1.004.27a4.497 4.497 0 0 0-.672-1.104c.493-.28 1.062-.451 1.709-.537.013.03.51 0 .6 0V1.5L12 3.429v.321a.327.327 0 0 1-.333.321h-3zM3.333 7.93c.369 0 .709-.103 1.004-.27.168.404.4.767.672 1.104-.493.28-1.062.451-1.676.537-.018-.011-2.435 0-2.666 0C.298 9.214 0 8.927 0 8.571c0-.355.298-.642.667-.642h2.666z"
                />
              </svg>
            </button>
            <div
              class="__react_component_tooltip tfc5d8380-d703-49b4-aa16-24435a79b198 place-top type-dark"
              data-id="tooltip"
              id="shuffle"
            >
              <style>
                
  	.tfc5d8380-d703-49b4-aa16-24435a79b198 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.tfc5d8380-d703-49b4-aa16-24435a79b198.place-top {
        margin-top: -10px;
    }
    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-top::before {
        border-top: 8px solid transparent;
    }
    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-bottom {
        margin-top: 10px;
    }
    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-left {
        margin-left: -10px;
    }
    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-left::before {
        border-left: 8px solid transparent;
    }
    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-right {
        margin-left: 10px;
    }
    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-right::before {
        border-right: 8px solid transparent;
    }
    .tfc5d8380-d703-49b4-aa16-24435a79b198.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
              </style>
              <span
                class="shuffle_tooltip"
              >
                Turn on Shuffle
              </span>
            </div>
          </div>
          <div
            class="control_volume_button"
          >
            <button
              class="volume_button"
              currentitem="false"
              data-for="volume"
              data-tip=""
            >
              <svg
                class="svg_volume"
                viewBox="0 0 32 32"
              >
                <path
                  d="M22.485 25.985c-0.384 0-0.768-0.146-1.061-0.439-0.586-0.586-0.586-1.535 0-2.121 4.094-4.094 4.094-10.755 0-14.849-0.586-0.586-0.586-1.536 0-2.121s1.536-0.586 2.121 0c2.55 2.55 3.954 5.94 3.954 9.546s-1.404 6.996-3.954 9.546c-0.293 0.293-0.677 0.439-1.061 0.439v0zM17.157 23.157c-0.384 0-0.768-0.146-1.061-0.439-0.586-0.586-0.586-1.535 0-2.121 2.534-2.534 2.534-6.658 0-9.192-0.586-0.586-0.586-1.536 0-2.121s1.535-0.586 2.121 0c3.704 3.704 3.704 9.731 0 13.435-0.293 0.293-0.677 0.439-1.061 0.439z"
                />
                <path
                  d="M13 30c-0.26 0-0.516-0.102-0.707-0.293l-7.707-7.707h-3.586c-0.552 0-1-0.448-1-1v-10c0-0.552 0.448-1 1-1h3.586l7.707-7.707c0.286-0.286 0.716-0.372 1.090-0.217s0.617 0.519 0.617 0.924v26c0 0.404-0.244 0.769-0.617 0.924-0.124 0.051-0.254 0.076-0.383 0.076z"
                />
              </svg>
            </button>
            <div
              class="__react_component_tooltip t9c11fe0b-d17e-4633-9e0a-347f29cc2f79 place-top type-dark allow_click"
              data-id="tooltip"
              id="volume"
            >
              <style>
                
  	.t9c11fe0b-d17e-4633-9e0a-347f29cc2f79 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-top {
        margin-top: -10px;
    }
    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-top::before {
        border-top: 8px solid transparent;
    }
    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-bottom {
        margin-top: 10px;
    }
    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-left {
        margin-left: -10px;
    }
    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-left::before {
        border-left: 8px solid transparent;
    }
    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-right {
        margin-left: 10px;
    }
    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-right::before {
        border-right: 8px solid transparent;
    }
    .t9c11fe0b-d17e-4633-9e0a-347f29cc2f79.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
              </style>
              <div
                class="input_check"
              >
                <input
                  max="10"
                  min="0"
                  step="1"
                  type="range"
                  value="5"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="next_queue"
        >
          <div
            class="queue_box"
          >
            <div
              class="queue_image"
              style="background-position: center; background-size: cover; background-repeat: no-repeat;"
            >
              <span>
                Queue
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
